File: src/test/java/com/freya02/botcommands/test/commands/slash/SlashCooldown.java
package com.freya02.botcommands.test.commands.slash;

import com.freya02.botcommands.api.commands.annotations.Cooldown;
import com.freya02.botcommands.api.commands.application.ApplicationCommand;
import com.freya02.botcommands.api.commands.application.slash.GuildSlashEvent;
import com.freya02.botcommands.api.commands.application.slash.annotations.JDASlashCommand;
import com.freya02.botcommands.api.commands.ratelimit.RateLimitScope;

import java.time.temporal.ChronoUnit;

public class SlashCooldown extends ApplicationCommand {
	@Cooldown(cooldown = 10, unit = ChronoUnit.SECONDS, rateLimitScope = RateLimitScope.USER)
	@JDASlashCommand(name = "cooldown", subcommand = "user")
	public void run(GuildSlashEvent event) {
		event.reply("ok").setEphemeral(true).queue();
	}

	@Cooldown(cooldown = 10, unit = ChronoUnit.SECONDS, rateLimitScope = RateLimitScope.GUILD)
	@JDASlashCommand(name = "cooldown", subcommand = "guild")
	public void run2(GuildSlashEvent event) {
		event.reply("ok2").setEphemeral(true).queue();
	}

	@Cooldown(cooldown = 10, unit = ChronoUnit.SECONDS, rateLimitScope = RateLimitScope.CHANNEL)
	@JDASlashCommand(name = "cooldown", subcommand = "channel")
	public void run3(GuildSlashEvent event) {
		event.reply("ok3").setEphemeral(true).queue();
	}
}

File: src/test/kotlin/io/github/freya022/botcommands/test_kt/commands/slash/SlashCooldown.kt
package io.github.freya022.botcommands.test_kt.commands.slash

import dev.minn.jda.ktx.messages.reply_
import io.github.freya022.botcommands.api.commands.annotations.Command
import io.github.freya022.botcommands.api.commands.annotations.Cooldown
import io.github.freya022.botcommands.api.commands.application.ApplicationCommand
import io.github.freya022.botcommands.api.commands.application.GlobalApplicationCommandManager
import io.github.freya022.botcommands.api.commands.application.annotations.AppDeclaration
import io.github.freya022.botcommands.api.commands.application.slash.GuildSlashEvent
import io.github.freya022.botcommands.api.commands.application.slash.annotations.JDASlashCommand
import io.github.freya022.botcommands.api.commands.builder.cooldown
import io.github.freya022.botcommands.api.commands.ratelimit.RateLimitScope
import java.time.temporal.ChronoUnit
import kotlin.time.Duration.Companion.seconds

@Command
class SlashCooldown : ApplicationCommand() {
    @JDASlashCommand(name = "cooldown_annotated")
    @Cooldown(cooldown = 5, unit = ChronoUnit.SECONDS, rateLimitScope = RateLimitScope.GUILD)
    fun onSlashCooldown(event: GuildSlashEvent) {
        event.reply_("ok", ephemeral = true).queue()
    }

    @AppDeclaration
    fun declare(globalApplicationCommandManager: GlobalApplicationCommandManager) {
        globalApplicationCommandManager.slashCommand("cooldown", function = ::onSlashCooldown) {
            cooldown(RateLimitScope.GUILD, 5.seconds)
        }
    }
}